### DISCUSSION TITLE
[Feature Request] A flag to enable/disable CropTransparentPixels

### DISCUSSION THREAD

**Discussion by chicknsoup** [AngusJohnson/Image32#44] [General]
Sometimes, I want to keep the original size of the loaded image (including the transparent part) so that every layer has the same size.
Is it possible to introduce a flag to TImage32 class to enable/disable the CropTransparentPixels api call?

**Comment by AngusJohnson** [comment]
I don't understand your question because you haven't explained the context in which CropTransparentPixels is being called.
However, I _suspect_ you're referring to the TRasterLayer32 class where layered raster images are automatically cropped. This I believe generally makes sense because it improves rendering performance (especially when these images are also transformed - rotated, skewed etc), and because these images will eventually be merged with other layers so I can't see any real benefit in retaining any surrounding transparency.

**Reply by chicknsoup** [reply]
Yes, I'm using TRasterLayer32. Let me explain my case:
- I have 4 layers (64x64 pixels each)
- The transparent edges of each layer are different

When loading each layer using TRasterLayer32.Image.LoadFromFile, the CropTransparentPixels will be called automatically, and that leads to an unexpected behavior where each layer has different size (not 64x64 anymore). In order to have the desired output image, I have to re-position and resize all 4 layers. 
Because the images are small, I dont see any performance issue with disabling CropTransparentPixels call.

**Comment by AngusJohnson** [comment]
OK, I'll have a closer look when I can. No promises 😁.
